specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json

```diff
@@ -1283,6 +1283,52 @@
           "application/json"
         ]
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Automanage/listServicePrincipalId": {
```

Comment:
>listServicePrincipalId [](http://example.com/codeflow?start=68&length=22)

Looks like this action returns a singleton, so the action name should be something more like getServicePrincipalId.

Reply:
Thank you for the comment!

I assume you are aware of the template function **listKeys** in Microsoft.OperationalInsights/workspaces (LogAnalytics).

This doc has the **list*** template functions by different RP: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list

Similar LogAnalytics **listKeys**, we are implementing **listServicePrincipalId** as template function.
If we use getServicePrincipalId then we cannot use as template function. Let us know If have further comments.

Reply:
I hadn't come across this before, thanks!